src/s3path.jl

```diff
@@ -333,14 +333,57 @@ function FilePathsBase.sync(f::Function, src::AbstractPath, dst::S3Path; delete=
     end
 end
 
+
+# for some reason, sometimes we get back a `Pair`
+# other times a `AbstractDict`.
+function _pair_or_dict_get(p::Pair, k)
+    first(p) == k || return nothing
+    return last(p)
+end
+_pair_or_dict_get(d::AbstractDict, k) = get(d, k, nothing)
+
+function _readdir_add_results!(results, r, key_length)
+    rm_key = s -> chop(s, head=key_length, tail=0)
+    sizehint!(results, length(results) + parse(Int, r["KeyCount"]))
+    if haskey(r, "CommonPrefixes")
```

Comment:
These two `if` statements are just duplicates, we can pull them into their own function. Maybe something like:

```julia
_retrieve_value(results, result_key, prefix_key)
```

Reply:
sure, done